docs/querying/api.md

```diff
@@ -1033,6 +1033,39 @@ $ curl http://localhost:9090/api/v1/status/tsdb
 
 *New in v2.15*
 
+### WAL Replay Stats
+
+The following endpoint returns information about the WAL replay:
+
+```
+GET /api/v1/status/walreplay
+```
+
+**read**: The number of segments replayed so far.
+**total**: The total number segments needed to be replayed.
+**progress**: The progress of the replay (0 - 100%).
+**state**: The state of the replay. Possible states:
+- **waiting**: Waiting for the replay to start.
+- **in progress**: The replay is in progress.
+- **done**: The replay has finished.
+
+```json
+$ curl http://localhost:9090/api/v1/status/walreplay
+{
+  "status": "success",
+  "data": {
+    "read": 2,
+    "total": 5
+    "progress": 40,
+    "state": "in progress"
```

Comment:
To ensure best API it's best to ensure the API is consistent and minimal. I still think it exposes too much information. As is there are multiple ways of getting the same information.

E.g to learn that the WAL replay is in progress I could:
* Check the percentage is higher than 0 and lower than 100
* I could check that `read` is higher than 0 and lower than `total`
* I could check the replay `state` == `in progress`

Having multiple ways of doing same thing for UI when talking to humans, sometimes makes sense, but for programmatic API it should be avoided. Every field adds extra cost. Adds and creates complexity, migration issues, documentation burden, support burden etc. 

Do you think we could aim for one minimal set of field that tells us everything?

I would really focus on literally one, two maybe 3 fields e.g:

```
status: "success", 
data:  {
  start_time: "<timestamp of replay start>"
  progress_percentage: 40,
  message: "replaying segment file 1", or doing X, doing Y, waiting for replay to start, parsing, finished - some human readable message

```

Reply:
message: "replaying segment file 1", or doing X, doing Y, waiting for replay to start, parsing, finished - some human readable message


Not really since we want to show a progressbar.

    "min": 210,
    "max": 271
    "current": 250,
    
   might be better

Reply:
Although `min`/`max`/`current` is a lot of fields, I think it's the best format because it provides machine-readable information about the magnitude of the replay not just for the starting screen but for other systems that might choose to process that data instead of just displaying it to a user (making the message less feasible). At the same time, I think we'll probably have to add in that state enum because checking those values alone isn't a reliable way to determine the status of the replay (if there are not segments, all three could be 0). I like the idea of `start_time` too, but maybe that's too many fields...

Reply:
So message is nice as it might tell what replay is focusing now, just nice to have. But yes not necessarily

Reply:
So should we add that on too, or just leave it at this?

Reply:
I would be against concrete number because:
* We may do things concurrently so suddenly number of segments will totally not aligned.
* We may do extra additional stages before, after or in between. 

I feel simple percentage allows us to decide on server side whatever percentage means and is the only thing that user needs as it's just loading bar, no?

Reply:
This API is not covered by our stability guarantees. I like the fact that the UI shows what we have in the logs.

Reply:
The think the one thing a percentage is missing is the magnitude of segments that need to be loaded. 40% could be 4/10 or 400/1000; the ladder will take significantly longer. Maybe we just include the total number of segments that need to be loaded, and leave out the min and current. I think it will be nice for a user to be able to estimate how much longer it will take. On the other hand, I do see your point with the alignment and extra stages.

Reply:
Cool, I think we find a good middleground (: